pandas/tests/io/json/test_pandas.py

```diff
@@ -117,6 +117,7 @@ def test_frame_non_unique_columns(self, orient, data):
                 expected.iloc[:, 0] = expected.iloc[:, 0].view(np.int64) // 1000000
         elif orient == "split":
             expected = df
+            expected.columns = ["x", "x.1"]
```

Comment:
should there also be a test for when the columns are a multiindex, the reader is `read_json`, and `orient='split'`?

Reply:
Good point. I'm not fully sure if it can be the case that `read_json(orient='split')` can generate a multiindex. What would the json look like?

I agree `dedup_names` can  probably be better tested, and make sure that the multiindex cases are well tested and work as expected. In my opinion probably better to address it in #50371. For this PR multiindexing is irrelevant if I'm not missing something. #50371 should consolidate the two different versions of `dedup_names` and make sure they work for all use cases if the implementations are not exactly the same. I think ensuring proper testing makes more sense there, here the function is not modified, and I don't think any bug in the function can be introduced in this PR, as it's just been moved from a method to a function.

Reply:
here's an example which passes on 1.5.2, but which fails here:
```python
data = [["a", "b"], ["c", "d"]]
df = DataFrame(data, index=[1, 2], columns=pd.MultiIndex.from_arrays([["x", "x"], ['a', 'x']]))
read_json(df.to_json(orient='split'), orient='split')
```

Reply:
Thanks @MarcoGorelli, this is very helpful. Looks like the json IO with `orient='split'` does try to support multiindexes, but the implementation in main seems buggy:

```python
>>> df = DataFrame(data, index=[1, 2], columns=pd.MultiIndex.from_arrays([["2022", "2022"], ['JAN', 'FEB']]))
>>> df
  2022    
   JAN FEB
1    a   b
2    c   d
>>> print(df.to_json(orient='split'))
{"columns":[["2022","JAN"],["2022","FEB"]],"index":[1,2],"data":[["a","b"],["c","d"]]}
>>> read_json(df.to_json(orient='split'), orient='split')
  2022 JAN
  2022 FEB
1    a   b
2    c   d
```

I opened #50456 to address the problem, and fixed the use case with multiindex and added the test, but xfailed it for now, so the current bug can be fixed in an separate PR.